examples/templates/gcp-linux/main.tf

```diff
@@ -70,21 +70,20 @@ resource "google_compute_instance" "dev" {
     email  = data.google_compute_default_service_account.default.email
     scopes = ["cloud-platform"]
   }
-  # The startup script runs as root with no $HOME environment set up, which can break workspace applications, so
-  # instead of directly running the agent init script, setup the home directory, write the init script, and then execute
-  # it.
+  # The startup script runs as root with no $HOME environment set up, so instead of directly 
+  # running the agent init script, create a user (with a homedir, default shell and sudo
+  # permissions) and execute the init script as that user.
   metadata_startup_script = <<EOMETA
 #!/usr/bin/env sh
 set -eux pipefail
 
-mkdir /root || true
-cat <<'EOCODER' > /root/coder_agent.sh
-${coder_agent.dev.init_script}
-EOCODER
-chmod +x /root/coder_agent.sh
-
-export HOME=/root
-/root/coder_agent.sh
-
+useradd -m -s /bin/bash "${local.linux_user}"
+echo "${local.linux_user} ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/coder-user
+exec sudo -u "${local.linux_user}" sh -c '${coder_agent.dev.init_script}'
```

Comment:
Since it's a persistent root disk, wouldn't this fail on subsequent boots since it doesn't check if the user already exists first?

Reply:
That's a good point actually Dean -- I didn't know this before, but apparently on [EC2 the user metadata script only runs on first boot by default](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html#user-data-cloud-init:~:text=are%20not%20run.-,User%20data%20and%20cloud%2Dinit%20directives,-The%20cloud%2Dinit), whereas on GCP it runs on every boot.

Reply:
Yeah, good catch. 

I originally wanted to just take the same approach that we use for AWS and create the user with cloud-init, which I now realize would have handled this. But unfortunately, cloud-init is excluded from the GCE variants of the Debian cloud images: https://salsa.debian.org/cloud-team/debian-cloud-images/-/issues/41